packages/yew-macro/src/html_tree/html_element.rs

```diff
@@ -247,27 +247,35 @@ impl ToTokens for HtmlElement {
                 }
             });
 
-            fn apply_as(directive: Option<&PropDirective>) -> TokenStream {
-                match directive {
-                    Some(PropDirective::ApplyAsProperty(token)) => {
-                        quote_spanned!(token.span()=> ::yew::virtual_dom::ApplyAttributeAs::Property)
-                    }
-                    None => quote!(::yew::virtual_dom::ApplyAttributeAs::Attribute),
-                }
-            }
-
             /// Try to turn attribute list into a `::yew::virtual_dom::Attributes::Static`
             fn try_into_static(
                 src: &[(LitStr, Value, Option<PropDirective>)],
             ) -> Option<TokenStream> {
+                if src
+                    .iter()
+                    .any(|(_, _, d)| matches!(d, Some(PropDirective::ApplyAsProperty(_))))
+                {
+                    // don't try to make a static attribute list if there are any properties to
+                    // assign
+                    return None;
+                }
                 let mut kv = Vec::with_capacity(src.len());
                 for (k, v, directive) in src.iter() {
                     let v = match v {
                         Value::Static(v) => quote! { #v },
                         Value::Dynamic(_) => return None,
                     };
-                    let apply_as = apply_as(directive.as_ref());
-                    kv.push(quote! { ( #k, #v, #apply_as ) });
+                    let v = match directive {
+                        Some(PropDirective::ApplyAsProperty(token)) => {
+                            quote_spanned!(token.span()=> ::yew::virtual_dom::AttributeOrProperty::Property(
+                                ::std::convert::Into::into(#v)
+                            ))
+                        }
+                        None => quote!(::yew::virtual_dom::AttributeOrProperty::Attribute(
+                            ::yew::virtual_dom::AttrValue::Static(#v)
+                        )),
```

Comment:
The CI failure on 1.64 is because of this. The borrow checker has improved and now allows this code that it previously didn't. We should increase the MSRV to take advantage of it

Reply:
What is the new MSRV?

I think we can bump the MSRV as long as it's not something like 1.70.

Reply:
1.72 is the last version that works with this

Reply:
@futursolo I worked around it, have a look